content/docs/learning-guide/tutorials/rendering/get-started-materialtypes-and-shaders.md

```diff
@@ -37,119 +35,134 @@ The `.material` file is created using **Material Editor** later in the tutorial.
 
 In this vertex shader program, you need the geometry's position. You can set up vertex input and output using the following code:
 
-```hlsl
+```cpp
 struct VertexInput
 {
     float3 m_position : POSITION;
 };
 
-struct VertexShaderOutput
+struct VertexOutput
 {
     float4 m_position : SV_Position;
 };
 ```
-The vertex shader program entry point is defined in the function `MainVS`. In this tutorial, you are creating a simple vertex shader that transforms the object's position from model space to clip space by applying a series of matrix transformations. 
 
-Your vertex shader program should look like this: 
-```hlsl
-VertexShaderOutput MainVS(VertexInput IN)
+The vertex shader program entry point is defined in the function `MainVS`. In this tutorial, you are creating a simple vertex shader that transforms the object's position from model space to clip space by applying a series of matrix transformations.
+
+Your vertex shader program should look like this:
+
+```cpp
+VertexOutput MainVS(VertexInput input)
 {
-    VertexShaderOutput OUT;
-    float3 worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1)).xyz;
-    OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, float4(worldPosition, 1.0));        
-    return OUT;
+    VertexOutput ouput;
+    const float4x4 worldMatrix = ObjectSrg::GetWorldMatrix();
+
+    float3 worldPosition = mul(worldMatrix, float4(input.m_position, 1.0)).xyz;
+    ouput.m_position = mul(ViewSrg::m_viewProjectionMatrix, float4(worldPosition, 1.0));
+
+    return ouput;
 }
 ```
+
 Fragment Shader
-: Next, set up the fragment input and output, and the fragment shader program. 
+: Next, set up the fragment input and output, and the fragment shader program.
 
-The fragment shader program entry point is defined in the function `MainPS`. It takes `VertexShaderOutput` as input. The resulting fragment shader outputs are stored in the `ForwardPassOutput` struct (included in `ForwardPassOutput.azsli`).
+The fragment shader program entry point is defined in the function `MainPS`. It takes `VertexOutput` as input. The resulting fragment shader outputs are stored in the `ForwardPassOutput` struct (included in `ForwardPassOutput.azsli`).
 
 Since this is an unlit single shader color, you only need to set the value for diffuse color and disable specular lighting.
 
-Your fragment shader program should look like this: 
-```hlsl
-ForwardPassOutput MainPS(VertexShaderOutput IN)
+Your fragment shader program should look like this:
+
+```cpp
+ForwardPassOutput MainPS(VertexOutput input)
 {
-    ForwardPassOutput OUT;
+    ForwardPassOutput ouput;
 
-    // standard ForwardPassOutput G-buffer layout
-    OUT.m_diffuseColor = float4(UnlitColorSrg::m_unlitColor, -1.0); // Subsurface scattering is disabled
-    OUT.m_specularColor = float4(0.0, 0.0, 0.0, 0.0);   // Disable specular lighting
+    ouput.m_diffuseColor = float4(UnlitColorSrg::m_unlitColor, -1.0);
```

Comment:
It can be confusing to see the -1 here without any explanation. You could restore the comment. Or do something like this:
```suggestion
    float subsurfaceScatteringDisabled = -1.0;
    ouput.m_diffuseColor = float4(UnlitColorSrg::m_unlitColor, subsurfaceScatteringDisabled);
```